libres/lib/job_queue/torque_driver.cpp

```diff
@@ -434,11 +435,27 @@ static int torque_driver_submit_shell_job(torque_driver_type *driver,
                 torque_debug(driver, "Submit arguments: %s",
                              stringlist_alloc_joined_string(remote_argv, " "));
                 char **argv = stringlist_alloc_char_ref(remote_argv);
-                int status = util_spawn_blocking(
-                    driver->qsub_cmd, stringlist_get_size(remote_argv),
-                    (const char **)argv, tmp_std_file, tmp_err_file);
-                if (status != 0) {
-                    torque_debug_spawn_status_info(driver, status);
+
+                /* The qsub command might fail intermittently for acceptable reasons,
+￼                  retry a couple of times with exponential sleep.  */
+                int return_value = -1;
+                int sleep_time = 2;
+                int max_sleep_time = 2 * 2 * 2 * 2; /* max 4 attempts */
```

Comment:
Is this like exponentially increasing `sleep_time`? If this is required then `max_sleep_time` could be just `max_sleep_count=4`, but I guess this doesn't require additional `i++`

Reply:
The exponentiality is covered by `sleep_time *= 2`. Testing `sleep_time < max_sleep_time` made the while-statement nicer as far as I remember, instead of having something like `max_sleep_count = 4`.